src/sentry/tasks/invalid_group_status_detection.py

```diff
@@ -0,0 +1,87 @@
+import random
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+from logging import getLogger
+
+from django.db import connection
+
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphistory import GroupHistoryStatus
+from sentry.silo import SiloMode
+from sentry.tasks.auto_ongoing_issues import TRANSITION_AFTER_DAYS
+from sentry.tasks.base import instrumented_task
+from sentry.types.group import SUBSTATUS_TO_STR, UNRESOLVED_SUBSTATUS_CHOICES, GroupSubStatus
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = getLogger(__name__)
+
+SAMPLE_SIZE = 500
+SEVEN_DAYS_AGO = datetime.now(tz=timezone.utc) - timedelta(days=TRANSITION_AFTER_DAYS)
+QUERY = """SELECT group_id, MAX(date_added)
+    FROM sentry_grouphistory
+    WHERE group_id IN ({group_ids}) and status = {status}
+    GROUP BY group_id
+"""
+
+
+@instrumented_task(
+    name="sentry.tasks.issues.invalid_group_status_detection",
+    queue="invalid_group_status_detection",
+    time_limit=5 * 60,  # 5 minutes
+    max_retries=3,
+    default_retry_delay=60,
+    silo_mode=SiloMode.REGION,
+)
+def detect_invalid_group_status() -> None:
+    group_ids = list(Group.objects.values_list("id", flat=True))
+    random_group_ids = random.sample(group_ids, SAMPLE_SIZE)
+    invalid_groups = defaultdict(list)
+    regressed_groups = []
+    escalating_groups = []
+    for group in RangeQuerySetWrapper(
```

Comment:
@snigdhas if we are only using a sample of 500 groups, we don't need to use `RangeQuerySetWrapper`